ntoskrnl/io/pnpmgr/pnproot.c

```diff
@@ -1473,7 +1473,7 @@ PnpRootPowerControl(
     Status = Irp->IoStatus.Status;
     IrpSp = IoGetCurrentIrpStackLocation(Irp);
 
-    if (DeviceExtension->Common.IsFDO)
+    if (DeviceExtension && DeviceExtension->Common.IsFDO)
```

Comment:
Looking at this and the comments from @Extravert-ir from #kernel-dev, it may be of interest to have an `ASSERT(DeviceExtension);` if the `DeviceExtension` actually must *not* be NULL there.

Reply:
Yeah, this doesn't make sense to me. The "pnp root" driver is in complete control over whether its device objects have a device extension (and the device extension is part of the same allocation as the device object, so there is no case where it's null due to OOM), and it can't get requests for device objects it doesn't own.
Something else is going on here...